cli/flox-rust-sdk/src/providers/flox_cpp_utils.rs

```diff
@@ -286,5 +311,26 @@ mod tests {
         );
     }
 
+    #[test]
+    fn catches_nix_errors() {
+        let (flox, _temp_dir) = flox_instance_with_optional_floxhub_and_client(None, true);
+        let manifest = formatdoc! {"
+        version =  1
+        "};
+        let mut env = new_path_environment(&flox, &manifest);
+        let pkgs = [PackageToInstall::Flake(FlakePackage {
+            id: "cachix".to_string(),
+            url: Url::parse("github:cachix/cachix").unwrap(),
```

Comment:
(curious, non-blocking): Will this always be broken? I guess we can't easily use a much simpler flake of our own because we don't allow local flake refs? I guess we could put something in `github.com/flox/flox` but changing it would be out-of-step with the code.

Reply:
Actually there's a new `_PKGDB_ALLOW_LOCAL_FLAKE` flag that we can use in the test that would allow us to support local flakes. I'd prefer to test against a tiny flake that does a minimal amount of eval.

Reply:
So I made a smaller test that takes about a second to run, but after that it's _super_ fast because it caches the eval failure lol

@mkenigs is there a way to prevent Nix from caching evaluation failures in tests?

Reply:
Why is it a problem to cache the eval failure?

Reply:
The error message changes from the actual error to the"cached eval failure of..." and becomes an impurity of the test.

Reply:
Ahh.. which is a reflection of the impurity of the function, rather than just the test, but only when used with local flakes and we're not exposing that to users.

Reply:
Can we just say either message is okay in the test? The user could get either behavior, so in that sense both messages are correct. I guess we could add an arg `pkgdb lock-flake-installable --no-eval-cache` but that doesn't seem worth it, since in general we always do want to use the eval cache

Reply:
Yeah either way it's testing the desired behavior, I just wanted to make it test one failure if possible.